src/utils/daos_hdlr.c

```diff
@@ -2643,42 +2636,63 @@ dm_connect(struct cmd_args_s *ap,
 					      "of the form:\n\t --dst </$pool> | </path/to/uns>");
 				D_GOTO(err, rc);
 			}
-			uuid_copy(ca->dst_c_uuid, dattr.da_cuuid);
+			snprintf(ca->dst_cont, DAOS_PROP_LABEL_MAX_LEN + 1, "%s", dattr.da_cont);
 		}
-		/* try to open container if this is a filesystem copy,
-		 * and if it fails try to create a destination,
-		 * then attempt to open again
+
+		/* try to open container if this is a filesystem copy, and if it fails try to create
+		 * a destination, then attempt to open again
 		 */
-		rc = daos_cont_open(ca->dst_poh, ca->dst_c_uuid, DAOS_COO_RW, &ca->dst_coh,
-				    dst_cont_info, NULL);
+		if (dst_cont_passed)
+			rc = daos_cont_open(ca->dst_poh, ca->dst_cont, DAOS_COO_RW, &ca->dst_coh,
+					    dst_cont_info, NULL);
+		else
+			rc = -DER_NONEXIST;
 		if (rc == -DER_NONEXIST) {
+			uuid_t cuuid;
+
 			if (ca->cont_layout == DAOS_PROP_CO_LAYOUT_POSIX) {
-				rc = dfs_cont_create(ca->dst_poh, ca->dst_c_uuid, NULL, NULL, NULL);
+				if (dst_cont_passed) {
+					rc = uuid_parse(ca->dst_cont, cuuid);
+					if (rc)
+						D_GOTO(err, rc);
+					rc = dfs_cont_create(ca->dst_poh, cuuid, NULL, NULL, NULL);
+				} else {
+					rc = dfs_cont_create(ca->dst_poh, &cuuid, NULL, NULL, NULL);
+					uuid_unparse(cuuid, ca->dst_cont);
+				}
 				if (rc != 0) {
 					rc = daos_errno2der(rc);
 					DH_PERROR_DER(ap, rc,
 						      "failed to create destination container");
 					D_GOTO(err, rc);
 				}
 			} else {
-				rc = daos_cont_create(ca->dst_poh, ca->dst_c_uuid, props, NULL);
+				if (dst_cont_passed) {
+					rc = uuid_parse(ca->dst_cont, cuuid);
+					if (rc == 0)
+						rc = daos_cont_create(ca->dst_poh, cuuid, props,
+								      NULL);
+					else
+						rc = daos_cont_create_with_label(ca->dst_poh,
+										 ca->dst_cont,
+										 props, NULL, NULL);
+				} else {
+					rc = daos_cont_create(ca->dst_poh, &cuuid, props, NULL);
+					uuid_unparse(cuuid, ca->dst_cont);
+				}
 				if (rc != 0) {
 					DH_PERROR_DER(ap, rc,
 						      "failed to create destination container");
 					D_GOTO(err, rc);
 				}
 			}
-			rc = daos_cont_open(ca->dst_poh, ca->dst_c_uuid, DAOS_COO_RW, &ca->dst_coh,
+			rc = daos_cont_open(ca->dst_poh, ca->dst_cont, DAOS_COO_RW, &ca->dst_coh,
 					    dst_cont_info, NULL);
 			if (rc != 0) {
 				DH_PERROR_DER(ap, rc, "failed to open container");
 				D_GOTO(err, rc);
 			}
-			fprintf(ap->outstream, "Successfully created container "DF_UUIDF"\n",
-				DP_UUID(ca->dst_c_uuid));
-		} else if (rc != 0) {
-			DH_PERROR_DER(ap, rc, "failed to open container");
-			D_GOTO(err, rc);
+			fprintf(ap->outstream, "Successfully created container %s\n", ca->dst_cont);
```

Comment:
@mchaarawi I am failing to see why this "successful" output was moved into the rc != 0 case?

Reply:
sorry i don't get your question.. where do you see rc != 0 with this success print?

Reply:
@mchaarawi yeah never mind I had to view the whole file since looking at it in the diff view I read it wrong.